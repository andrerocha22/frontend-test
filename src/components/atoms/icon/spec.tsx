import React from 'react';
import { shallow, ShallowWrapper, mount } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import 'jest-styled-components';

import { Icon } from './index';

const defaultProps = {
  color: '#ffffff',
  backgroundColor: '#000000',
  src: 'iconTest',
};

const mockCallBack = jest.fn();

const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<Icon clickCallback={mockCallBack} {...setupProps} />);
};

describe('test icon', () => {
  let component: ShallowWrapper = undefined;

  test('renders without error', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-icon');
    expect(component.length).toBe(1);
  });

  test('card snapshot test ', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-icon');
    expect(component).toMatchSnapshot();
  });

  test('test icon click', () => {
    const wrapper = setup();
    component = findByTestAttr(wrapper, 'component-icon');
    component.simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });
});
