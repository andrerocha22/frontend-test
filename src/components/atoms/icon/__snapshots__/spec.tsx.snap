// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test icon card snapshot test  1`] = `
<styled.img
  backgroundColor="#000000"
  clickCallback={[MockFunction]}
  color="#ffffff"
  data-test="component-icon"
  onClick={[MockFunction]}
  src="iconTest"
/>
`;
